src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java

```diff
@@ -38,85 +38,75 @@
     @Autowired
     private UserProfileClient userProfileClient;
 
-    @Autowired
-    private SyncJobRepository syncJobRepository;
-
     @Value("${loggingComponentName}")
-    protected String loggingComponentName;
+    private String loggingComponentName;
 
-    public void updateUserProfile(String searchQuery, String bearerToken, String s2sToken, List<IdamClient.User> users)
+    public ProfileSyncAudit updateUserProfile(String searchQuery, String bearerToken, String s2sToken,
+                                              Set<IdamClient.User> users, ProfileSyncAudit syncAudit)
             throws UserProfileSyncException {
-        log.info(loggingComponentName, "Inside updateUserProfile:: ");
+        log.info("{}:: Inside updateUserProfile::{} ", loggingComponentName);
+        List<ProfileSyncAuditDetails> profileSyncAuditDetails = new ArrayList<>();
         users.forEach(user -> {
-            Optional<GetUserProfileResponse> userProfile = userAcquisitionService.findUser(bearerToken,
-                    s2sToken, user.getId());
+            Optional<GetUserProfileResponse> userProfile = userAcquisitionService.findUser(bearerToken, s2sToken,
+                    user.getId());
 
             if (userProfile.isPresent()) {
-                Map<String, Boolean> status = new HashMap<String, Boolean>();
-                status.put(IdamStatus.ACTIVE.name(), user.isActive());
-                status.put(IdamStatus.PENDING.name(), user.isPending());
+                StringBuilder sb = new StringBuilder();
+                sb.append(user.isActive());
+                sb.append(user.isPending());
                 UserProfile updatedUserProfile = UserProfile.builder()
```

Comment:
is nt this mutually exclusive?  why is this required?

Reply:
idam some times returns both flags in the response so to support that added above condition